packages/apidom-ls/src/services/validation/linter-functions.ts

```diff
@@ -1068,4 +1069,41 @@ export const standardLinterfunctions: FunctionItem[] = [
       return true;
     },
   },
+  {
+    functionName: 'apilintOpenAPIParameterFieldIsDefinedWithinPathTemplate',
+    function: (element: Element) => {
+      if (element.element === 'parameter') {
+        const parameterLocation = toValue((element as ObjectElement).get('in'));
+        const isInPathItemElement =
+          isArrayElement(element.parent) &&
+          includesClasses(['path-item-parameters'], element.parent);
+
+        if (!isInPathItemElement || parameterLocation !== 'path') {
+          return true;
+        }
+
+        const pathItemElement = element.parent.parent.parent;
+        const isPathItemPartOfPathTemplating = isStringElement(pathItemElement.meta.get('path'));
+
+        if (!isPathItemPartOfPathTemplating) {
+          return true;
+        }
+
+        const pathTemplate = toValue(pathItemElement.meta.get('path'));
+        const parameterName = toValue((element as ObjectElement).get('name'));
+
+        const parseResult = parse(pathTemplate);
+        const parts: [string, string][] = [];
+        parseResult.ast.translate(parts);
+
+        const pathTemplateASTIncludesParameter = (ast: [string, string][]) =>
```

Comment:
Do we need to define an ad-hoc function here? It looks like a lot of code that can be simplified a bit. Here is a suggestion:

```js
const parseResult = parse(pathTemplate);
const parts: [string, string][] = [];

parseResult.ast.translate(parts);
return parts.some(([name, value]) => name === 'template-expression-param-name' && value === parameterName);
```